### ISSUE TITLE
Data binding won't use custom validator

### ISSUE THREAD

**Issue by pohanlSUM** [fyne-io/fyne#1969]
### Describe the bug:
When use data binding, fyne will use its own validator instead of the custom validator set by someEntry.Validator = xxx

### To Reproduce:
See example code

### Screenshots:
See example code

### Example code:
```go
someStr := binding.NewString()
someEntry := widget.NewEntryWithData(someStr)
someEntry.SetPlaceHolder("Placeholder")
someEntry.Validator = someValidator()
```
In this condition, although someValidator() will be called, but the error returned by someValidator() will be overwritten by
```go
convertErr = data.Set(s)
e.SetValidationError(convertErr)
```
Workaround:
```go
var onChangedCallBack func(string)
onChangedCallBack = someEntry.OnChanged
someEntry.OnChanged = func(text string) {
	onChangedCallBack(text)
	someEntry.SetValidationError(someEntry.Validate())
}
```

### Device (please complete the following information):
 - **OS:** Ubuntu
 - **Version:** 18
 - **Go version:** 15.5
 - **Fyne version:** 2.0

**Comment by andydotxyz** [comment]
Thanks, this probably needs tidied up.
We had intended that validation would be in the data binding like #1704 but that currently requires developer code as there is no helper functions.

**Comment by pohanlSUM** [comment]
> Thanks, this probably needs tidied up.
> We had intended that validation would be in the data binding like #1704 but that currently requires developer code as there is no helper functions.

I think this would be a solution.
In the mean time, I think data binding should not be implemented in OnChanged callback, if user want to customize OnChanged callback function, it will overwrite data binding

**Comment by andydotxyz** [comment]
That is probably true, we had assumed that data binding would be the source of truth and so it took control of `OnChanged` and `Validator`.
Keeping this open as we could do better.

**Comment by AlbinoGeek** [comment]
What if `XWithData` hooked up the `Validator`?

It seems like this might result in a much cleaner API.

You would still assign validators to the widgets (e.g: no change here.)

But, the `XWithData` should hook up the validator utilizing Data Binding, thoughts?

**Comment by andydotxyz** [comment]
The `...WithData` does hook up validation, that is what the problem is here...
The OP wants to have data binding *and* a separate validator.

**Comment by andydotxyz** [comment]
I wonder if there may be a way around this - if the thing hooking in validation didn't push the value in directly but instead made it available through the usual validation calls

**Comment by andydotxyz** [comment]
This was resolved by 77db0b32 I am pretty sure.
